### ISSUE TITLE
[boost-random] Build error (x64-UWP)

### ISSUE THREAD

**Issue by danilogr** [microsoft/vcpkg#26996]
Package: boost-random[core]:x64-uwp -> 1.80.0

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.29.30146.0
-    vcpkg-tool version: 2022-09-20-522aa94e9d261c7d7b2f079bf2591ca62df5c714
    vcpkg-scripts version: 9d0351aed 2022-09-26 (70 minutes ago)

**To Reproduce**

`vcpkg install boost:x64-uwp`

**Failure logs**

```
-- Using cached boostorg-random-boost-1.80.0.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/boost-random/src/ost-1.80.0-78c2eef13e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/boostorg-random-boost-1.80.0.tar.gz
-- Using source at C:/vcpkg/buildtrees/boost-random/src/ost-1.80.0-78c2eef13e.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x64-uwp
-- Configuring x64-uwp
-- Building x64-uwp-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:129 (message):
    Command failed: C:/vcpkg/downloads/tools/cmake-3.24.0-windows/cmake-3.24.0-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j25
    Working Directory: C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\boost-random\install-x64-uwp-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  installed/x64-windows/share/boost-build/boost-modular-build.cmake:136 (vcpkg_cmake_install)
  ports/boost-random/portfile.cmake:12 (boost_modular_build)
  scripts/ports.cmake:147 (include)



```
<details><summary>C:\vcpkg\buildtrees\boost-random\install-x64-uwp-dbg-out.log</summary>

```
[1/2] cmd.exe /C "cd /D C:\vcpkg\buildtrees\boost-random\src\ost-1.80.0-78c2eef13e.clean\build && C:\vcpkg\installed\x64-windows\tools\boost-build\b2.exe toolset=msvc --user-config=C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/user-config.jam --stagedir=C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/stage --build-dir=C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg runtime-link=shared link=shared address-model=64 architecture=x86 target-os=windows windows-api=store linkflags=WindowsApp.lib threadapi=win32 variant=debug --disable-icu --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -j24 -sBOOST_ROOT=C:/vcpkg/installed/x64-windows/tools/boost-build -sBOOST_BUILD_PATH=C:/vcpkg/installed/x64-windows/tools/boost-build --debug-configuration --debug-building --debug-generators --ignore-site-config --hash -q debug-symbols=on -d +2 threading=multi stage"
FAILED: CMakeFiles/boost C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/CMakeFiles/boost 
cmd.exe /C "cd /D C:\vcpkg\buildtrees\boost-random\src\ost-1.80.0-78c2eef13e.clean\build && C:\vcpkg\installed\x64-windows\tools\boost-build\b2.exe toolset=msvc --user-config=C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/user-config.jam --stagedir=C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/stage --build-dir=C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg runtime-link=shared link=shared address-model=64 architecture=x86 target-os=windows windows-api=store linkflags=WindowsApp.lib threadapi=win32 variant=debug --disable-icu --with-atomic --with-random --with-date_time --with-filesystem --with-system --with-thread --with-chrono -j24 -sBOOST_ROOT=C:/vcpkg/installed/x64-windows/tools/boost-build -sBOOST_BUILD_PATH=C:/vcpkg/installed/x64-windows/tools/boost-build --debug-configuration --debug-building --debug-generators --ignore-site-config --hash -q debug-symbols=on -d +2 threading=multi stage"
notice: loading B2 from C:/vcpkg/installed/x64-windows/tools/boost-build/src/kernel/bootstrap.jam
notice: Site configuration files will be ignored due to the
notice: --ignore-site-config command-line option.
notice: Loading explicitly specified user configuration file:
    C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\user-config.jam
notice: Searching 'C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg' for user-config configuration file 'user-config.jam'.
notice: Loading user-config configuration file 'user-config.jam' from 'C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg'.
notice: [msvc-cfg] msvc-14.2 detected, command: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.16.27023\bin\HostX64\arm\cl.exe'
notice: will use 'C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe' for msvc, condition <toolset>msvc-142
notice: [msvc-cfg] condition: '<toolset>msvc-142/<architecture>x86/<address-model>32', setup: 'C:/vcpkg/installed/x64-windows/share/boost-build/nothing.bat x86'
notice: [msvc-cfg] condition: '<toolset>msvc-142/<architecture>/<address-model>32', setup: 'C:/vcpkg/installed/x64-windows/share/boost-build/nothing.bat x86'
...
Skipped 129 lines
...

     Usage requirements for stage:  <define>BOOST_RANDOM_DYN_LINK <relevant>address-model <relevant>address-sanitizer <relevant>architecture <relevant>asynch-exceptions <relevant>cxxstd <relevant>cxxstd-dialect <relevant>debug-store <relevant>debug-symbols <relevant>deduced-address-model <relevant>deduced-architecture <relevant>define:<relevant>link <relevant>embed-manifest <relevant>embed-manifest-via <relevant>exception-handling <relevant>extern-c-nothrow <relevant>inlining <relevant>instruction-set <relevant>link <relevant>lto <relevant>lto-mode <relevant>optimization <relevant>pch <relevant>rtti <relevant>runtime-debugging <relevant>runtime-link <relevant>segmented-stacks <relevant>threading <relevant>toolset <relevant>toolset-msvc:vendor <relevant>toolset-msvc:version <relevant>variant <relevant>warnings <relevant>warnings-as-errors <relevant>windows-api <xdll-path>/C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/boost/build/9357143429b3339f991610d5cdc76bfc
     Build properties:  <address-model>64 <architecture>x86 <archiveflags>-machine:x64 -nologo <asynch-exceptions>off <coverage>off <debug-store>object <debug-symbols>on <define>BOOST_ALL_NO_LIB=1 <define>BOOST_RANDOM_DYN_LINK <define>WINAPI_FAMILY=WINAPI_FAMILY_APP <embed-manifest-via>linker <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <hardcode-dll-paths>false <host-os>windows <include>../include&&/C:/vcpkg/installed/x64-uwp/include <inlining>off <install-dependencies>off <link>shared <linkflags>-MANIFEST:NO -NXCOMPAT -DYNAMICBASE -DEBUG -WINMD:NO -APPCONTAINER -SUBSYSTEM:CONSOLE -MANIFESTUAC:NO -nologo -debug <linkflags>/APPCONTAINER <linkflags>WindowsApp.lib <location>/C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/stage/lib <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>address-model <relevant>address-sanitizer <relevant>architecture <relevant>asynch-exceptions <relevant>cxxstd <relevant>cxxstd-dialect <relevant>debug-store <relevant>debug-symbols <relevant>deduced-address-model <relevant>deduced-architecture <relevant>define:<relevant>link <relevant>embed-manifest <relevant>embed-manifest-via <relevant>exception-handling <relevant>extern-c-nothrow <relevant>inlining <relevant>instruction-set <relevant>link <relevant>lto <relevant>lto-mode <relevant>optimization <relevant>pch <relevant>rtti <relevant>runtime-debugging <relevant>runtime-link <relevant>segmented-stacks <relevant>threading <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <relevant>toolset <relevant>toolset-msvc:vendor <relevant>toolset-msvc:version <relevant>variant <relevant>warnings <relevant>warnings-as-errors <relevant>windows-api <response-file>auto <rtti>on <runtime-debugging>on <runtime-link>shared <stdlib>native <strip>off <suppress-import-lib>false <symlink-location>project-relative <tag>@Jamfile<C:\vcpkg\buildtrees\boost-random\src\ost-1.80.0-78c2eef13e.clean>%Jamfile<C:\vcpkg\buildtrees\boost-random\src\ost-1.80.0-78c2eef13e.clean>.tag <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>142 <toolset>msvc <translate-path>@Jamfile<C:\vcpkg\buildtrees\boost-random\src\ost-1.80.0-78c2eef13e.clean\build>%Jamfile<C:\vcpkg\buildtrees\boost-random\src\ost-1.80.0-78c2eef13e.clean\build>.stage-translate-path <user-interface>console <variant>debug <vectorize>off <warnings-as-errors>off <warnings>on <windows-api>store <xdll-path>/C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/boost/build/9357143429b3339f991610d5cdc76bfc
     *** construct INSTALLED_SHARED_LIB
         from { msvc%msvc.link.dll-boost_random-vc142-mt-gd-x64-1_80.dll.SHARED_LIB { msvc%msvc.compile.c++-random_device.obj.OBJ { random_device.cpp.CPP } } }
         properties: <address-model>64 <architecture>x86 <archiveflags>-machine:x64 -nologo <asynch-exceptions>off <coverage>off <debug-store>object <debug-symbols>on <define>BOOST_ALL_NO_LIB=1 <define>BOOST_RANDOM_DYN_LINK <define>WINAPI_FAMILY=WINAPI_FAMILY_APP <embed-manifest-via>linker <embed-manifest>on <exception-handling>on <extern-c-nothrow>off <hardcode-dll-paths>true <host-os>windows <include>../include&&/C:/vcpkg/installed/x64-uwp/include <inlining>off <install-dependencies>off <link>shared <linkflags>-MANIFEST:NO -NXCOMPAT -DYNAMICBASE -DEBUG -WINMD:NO -APPCONTAINER -SUBSYSTEM:CONSOLE -MANIFESTUAC:NO -nologo -debug <linkflags>/APPCONTAINER <linkflags>WindowsApp.lib <location>/C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/stage/lib <main-target-type>LIB <midl-robust>yes <midl-stubless-proxy>yes <optimization>off <os>NT <pch>on <preserve-test-targets>on <profiling>off <relevant>address-model <relevant>address-sanitizer <relevant>architecture <relevant>asynch-exceptions <relevant>cflags <relevant>compileflags <relevant>cxxflags <relevant>cxxstd <relevant>debug-store <relevant>debug-symbols <relevant>def-file <relevant>define <relevant>define:<relevant>link <relevant>embed-manifest <relevant>embed-manifest-file <relevant>embed-manifest-via <relevant>exception-handling <relevant>extern-c-nothrow <relevant>find-shared-library <relevant>find-static-library <relevant>force-include <relevant>include <relevant>inlining <relevant>instruction-set <relevant>library-file <relevant>library-path <relevant>link <relevant>linkflags <relevant>lto <relevant>main-target-type <relevant>mc-input-encoding <relevant>mc-output-encoding <relevant>mc-set-customer-bit <relevant>optimization <relevant>pch <relevant>pch-file <relevant>pch-header <relevant>pch-source <relevant>response-file <relevant>rtti <relevant>runtime-debugging <relevant>runtime-link <relevant>suppress-import-lib <relevant>threading <relevant>threading:<relevant>runtime-link <relevant>threading:<relevant>toolset <relevant>toolset <relevant>toolset-msvc:version <relevant>undef <relevant>user-interface <relevant>warnings <relevant>warnings-as-errors <relevant>windows-api <response-file>auto <rtti>on <runtime-debugging>on <runtime-link>shared <stdlib>native <strip>off <suppress-import-lib>false <symlink-location>project-relative <target-os>windows <testing.execute>on <threadapi>win32 <threading>multi <toolset-msvc:version>142 <toolset>msvc <user-interface>console <variant>debug <vectorize>off <warnings-as-errors>off <warnings>on <windows-api>store
     find-viable-generators target-type= INSTALLED_SHARED_LIB property-set= msvc-142/debug/address-model-64/architecture-x86/threading-multi/windows-api-store
     trying type INSTALLED_SHARED_LIB
     there are generators for this type
     trying generator install-shared-lib ( SHARED_LIB -> INSTALLED_SHARED_LIB )
       is viable
     ***  1  viable generators
     returned { common%common.copy-boost_random-vc142-mt-gd-x64-1_80.dll.SHARED_LIB { msvc%msvc.link.dll-boost_random-vc142-mt-gd-x64-1_80.dll.SHARED_LIB { msvc%msvc.compile.c++-random_device.obj.OBJ { random_device.cpp.CPP } } } }
       generator install-shared-lib  spawned 
       { common%common.copy-boost_random-vc142-mt-gd-x64-1_80.dll.SHARED_LIB { msvc%msvc.link.dll-boost_random-vc142-mt-gd-x64-1_80.dll.SHARED_LIB { msvc%msvc.compile.c++-random_device.obj.OBJ { random_device.cpp.CPP } } } }
       with usage requirements: [ ]
     Usage requirements from stage:  <define>BOOST_RANDOM_DYN_LINK <relevant>address-model <relevant>address-sanitizer <relevant>architecture <relevant>asynch-exceptions <relevant>cxxstd <relevant>cxxstd-dialect <relevant>debug-store <relevant>debug-symbols <relevant>deduced-address-model <relevant>deduced-architecture <relevant>define:<relevant>link <relevant>embed-manifest <relevant>embed-manifest-via <relevant>exception-handling <relevant>extern-c-nothrow <relevant>inlining <relevant>instruction-set <relevant>link <relevant>lto <relevant>lto-mode <relevant>optimization <relevant>pch <relevant>rtti <relevant>runtime-debugging <relevant>runtime-link <relevant>segmented-stacks <relevant>threading <relevant>toolset <relevant>toolset-msvc:vendor <relevant>toolset-msvc:version <relevant>variant <relevant>warnings <relevant>warnings-as-errors <relevant>windows-api <xdll-path>/C:/vcpkg/buildtrees/boost-random/x64-uwp-dbg/boost/build/9357143429b3339f991610d5cdc76bfc
msvc.read-setup <pC:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3>msvc-setup.read
 
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage\lib

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage\lib\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage\lib"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc"
    
common.mkdir C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3

        if not exist "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3\\" mkdir "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3"
    

file C:\Users\dnlme\AppData\Local\Temp\jam4988ed6f.000

msvc.write-setup-script C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3\msvc-setup.bat

    type "C:\Users\dnlme\AppData\Local\Temp\jam4988ed6f.000" > "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3\msvc-setup.bat"

compile-c-c++ C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc\random_device.obj

    call "C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\standalone\msvc\6b07b0b0f8e08318738a96bc606093b3\msvc-setup.bat" amd64 >nul
 cl "..\src\random_device.cpp" -c -Fo"C:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc\random_device.obj"     -TP /wd4675 /EHs /GR /Zc:throwingNew -DWIN32 -D_WINDOWS -D_UNICODE -DUNICODE -DWINAPI_FAMILY=WINAPI_FAMILY_APP -D__WRL_NO_DEFAULT_LIB__ -nologo -Z7 -MP -GS -Gd -Gm- -W3 -WX- -Zc:wchar_t -Zc:inline -Zc:forScope -fp:precise -Oy- -EHsc -utf-8 -D_DEBUG -Od -RTC1 -MDd -ZW:nostdlib /AI"C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs\Microsoft.VCLibs\14.0\References\CommonConfiguration\neutral" /AI"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x86\store\references" /AI"C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.22000.0" /AI"C:\Program Files (x86)\Windows Kits\10\References\10.0.22000.0" /AI"C:\Windows\Microsoft.NET\Framework64\v4.0.30319" /D_WIN32_WINNT=0x0A00 /Z7 /Od /Ob0 /W3 /MDd /Zc:forScope /Zc:wchar_t /Zc:inline /favor:blend -DBOOST_ALL_NO_LIB=1 -DBOOST_RANDOM_DYN_LINK -DWINAPI_FAMILY=WINAPI_FAMILY_APP "-I..\include" "-IC:\vcpkg\installed\x64-uwp\include" 

random_device.cpp
..\src\random_device.cpp(95): error C3861: 'CryptEnumProvidersA': identifier not found
..\src\random_device.cpp(104): error C3861: 'CryptAcquireContextA': identifier not found
..\src\random_device.cpp(121): error C3861: 'CryptGenRandom': identifier not found
...skipped <pC:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc>boost_random-vc142-mt-gd-x64-1_80.dll for lack of <pC:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc>random_device.obj...
...skipped <pC:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\stage\lib>boost_random-vc142-mt-gd-x64-1_80.dll for lack of <pC:\vcpkg\buildtrees\boost-random\x64-uwp-dbg\boost\build\9357143429b3339f991610d5cdc76bfc>boost_random-vc142-mt-gd-x64-1_80.dll...
...failed updating 1 target...
ninja: build stopped: subcommand failed.
```
</details>
**Additional context**

In short, this seems to be the issue
[install-x64-uwp-dbg-out.log](https://github.com/microsoft/vcpkg/files/9648899/install-x64-uwp-dbg-out.log)

```random_device.cpp
..\src\random_device.cpp(95): error C3861: 'CryptEnumProvidersA': identifier not found
..\src\random_device.cpp(104): error C3861: 'CryptAcquireContextA': identifier not found
..\src\random_device.cpp(121): error C3861: 'CryptGenRandom': identifier not found
```

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. Store(UWP) do not have WinCrypt libraries, so boost-random does not support uwp, this is a upstream issue. You can refer to this comment from upstream PR: https://github.com/boostorg/random/pull/29#issuecomment-334961290. You can submit an issue to upstream.

**Comment by LilyWangLL** [comment]
Since this issue is a upstream issue and need upstream to resolve this issue, so I am closing this issue for now. If the upstream has a solution to resolve this issue, please reopen this issue.
